crates/ty_python_semantic/src/types/class.rs

```diff
@@ -1289,6 +1289,14 @@ impl<'db> ClassLiteral<'db> {
 
                 Some(Type::Callable(CallableType::single(db, signature)))
             }
+            (CodeGeneratorKind::NamedTuple, name) if name != "__init__" => {
```

Comment:
I needed to exclude `__init__` here (for now) because we currently try to call it during `NamedTuple` construction (which is probably wrong?):
```py
class NamedTupleFallback(tuple[Any, ...]):
    # …
    @overload
    def __init__(self, typename: str, fields: Iterable[tuple[str, Any]], /) -> None: ...
    @overload
    @typing_extensions.deprecated(
        "Creating a typing.NamedTuple using keyword arguments is deprecated and support will be removed in Python 3.15"
    )
    def __init__(self, typename: str, fields: None = None, /, **kwargs: Any) -> None: ...
```

Reply:
That was wrong/imprecise. I should have said: it's wrong that we try to call *this specific* `__init__` method.

At runtime, there's no `__init__` method on named tuples:
```pycon
>>> from typing import NamedTuple
>>> class N(NamedTuple):
...     name: str
...     
>>> N.__new__
<function N.__new__ at 0x7d79400b2020>
>>> N.__init__
<slot wrapper '__init__' of 'object' objects>
```

Instead, at runtime, `typing.NamedTuple` is a *function* with the signature of that `NamedTupleFallback.__init__` method. It might help us model a `NamedTuple("N", ["foo", "bar"])` call, but I don't see how it would help to include that as a method on user-generated `NamedTuple`s.